### PR TITLE
#230: Created new deploy method with similar parameters as in python-extension-common

### LINKED ISSUE #230: Align parameter for `upload` command with PEC
## Background

See https://github.com/exasol/script-languages-container-tool/pull/228#discussion_r1733970823

### PR DESCRIPTION
**Pull request by tomuben**
fixes #230

### CODE REVIEW

File: exasol_script_languages_container_tool/cli/commands/deploy.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    f"--{SecretParams.BUCKETFS_PASSWORD.value}",
+    type=str,
+    prompt="BucketFS password",
+    prompt_required=False,
+    hide_input=True,
+    default=SECRET_DISPLAY,
+    callback=secret_callback,
+)
+@click.option("--path-in-bucket", type=str, required=False, default="")
+@add_options(release_options)  # 1
```
Comment (tkilias) [inline exasol_script_languages_container_tool/cli/commands/deploy.py:81]: What are the comments with the numbers behind the lines

File: exasol_script_languages_container_tool/cli/commands/deploy.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+@click.option("--release-name", type=str, default=None)
+@add_options(build_options)  # 7
+@add_options(docker_repository_options)  # 4
+@add_options(system_options)  # 2
+@add_options(luigi_logging_options)  # 2
+@click.option("--ssl-cert-path", type=str, default="")
```
Comment (tkilias) [inline exasol_script_languages_container_tool/cli/commands/deploy.py:87]: I think, these should move upwards to the bucketfs parameters

File: exasol_script_languages_container_tool/cli/commands/upload.py [hunk 3]
```diff
@@ -26,7 +26,9 @@
 from exasol_script_languages_container_tool.lib import api
 
 
-@cli.command(short_help="Uploads the script-language-container to the database.")
+@cli.command(
+    short_help="Uploads the script-language-container to the database.Deprecated."
```
Comment (tkilias) [inline exasol_script_languages_container_tool/cli/commands/upload.py:30]:
```suggestion
    short_help="Uploads the script-language-container to the database. [Deprecated]"
```

File: exasol_script_languages_container_tool/lib/api/upload.py
Comment (tkilias) [inline exasol_script_languages_container_tool/lib/api/upload.py:1]: We should actually throw a warning that this function is deprecated

File: test/test_docker_deploy.py
Comment (tkilias) [inline test/test_docker_deploy.py:1]: API test missing

Comment (tkilias) [reply test/test_docker_deploy.py:1]: Or better for functionality we should start with the API test and implement the cli as a mocked one using clickrunner

File: exasol_script_languages_container_tool/cli/commands/deploy.py (hunk 1 above)
Comment (tomuben) [reply exasol_script_languages_container_tool/cli/commands/deploy.py:81]: Ups...an artificat which should not have been pushed to Github. I was trying to debug why the `test_click_api_consistency.py` test failed

Comment (tomuben) [reply exasol_script_languages_container_tool/cli/commands/deploy.py:81]: removed it

File: exasol_script_languages_container_tool/cli/commands/deploy.py (hunk 2 above)
Comment (tomuben) [reply exasol_script_languages_container_tool/cli/commands/deploy.py:87]: Done

File: exasol_script_languages_container_tool/cli/commands/upload.py (hunk 3 above)
Comment (tomuben) [reply exasol_script_languages_container_tool/cli/commands/upload.py:30]: Done

File: test/test_docker_deploy.py
Comment (tomuben) [reply test/test_docker_deploy.py:1]: Done

File: exasol_script_languages_container_tool/lib/api/upload.py [hunk 4]
```diff
@@ -55,13 +56,9 @@ def upload(
     ssl_cert_path: str = "",
     use_ssl_cert_validation: bool = True,
 ) -> luigi.LocalTarget:
-    """
-    This command uploads the whole script-language-container package of the flavor to the database.
-    If the stages or the packaged container do not exists locally, the system will build, pull or
-    export them before the upload.
-    :raises api_errors.TaskFailureError: if operation is not successful.
-    :return: Path to resulting report file.
-    """
```
Comment (tkilias) [inline exasol_script_languages_container_tool/lib/api/upload.py:64]: documentation should stay anyway I think

File: test/unit/cli/test_deploy.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+
+@pytest.fixture
+def cli():
+    """
+    To prevent accidentally creating actual AWS resources, the fixture
+    tells the CliRunner to use an invalid AWS profile.
```
Comment (tkilias) [inline test/unit/cli/test_deploy.py:26]: copy paste

File: test/unit/cli/test_deploy.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    return_mock = MagicMock()
+    dummy_returned_target = DummyLocalTarget()
+    return_mock.open.return_value = dummy_returned_target
+
+    with patch(
+        "exasol_script_languages_container_tool.lib.api.deploy",
+        return_value=return_mock,
+    ) as mock_foo:
+        with tempfile.TemporaryDirectory() as temp_flavor_path:
+            cli.run(
```
Comment (tkilias) [inline test/unit/cli/test_deploy.py:45]: maybe the constants your putting and against you compare should be defined as constants

File: test/unit/cli/test_deploy.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    return_mock.open.assert_called_once()
+    dummy_returned_target.mock.read.assert_called_once()
+
+
+def test_deploy_password_in_env(cli):
+    return_mock = MagicMock()
+    dummy_returned_target = DummyLocalTarget()
+    return_mock.open.return_value = dummy_returned_target
+
+    cli.env = {"BUCKETFS_PASSWORD": "super_secret_bucketfs_password"}
```
Comment (tkilias) [inline test/unit/cli/test_deploy.py:106]: same super_secret_bucketfs_password probably should be a constant

File: test/unit/test_click_api_consistency.py
Comment (tkilias) [inline test/unit/test_click_api_consistency.py:1]: weird that he doesn't recognize the move, did this file change besides the moving

File: exasol_script_languages_container_tool/lib/api/upload.py (hunk 4 above)
Comment (tomuben) [reply exasol_script_languages_container_tool/lib/api/upload.py:64]: The Python standard libraries remove the documentation when a method becomes _deprecated_:
https://github.com/python/cpython/blob/8aaf7525ab839c32966ee862363ad2543a721306/Lib/logging/__init__.py#L1534

File: test/unit/cli/test_deploy.py (hunk 5 above)
Comment (tomuben) [reply test/unit/cli/test_deploy.py:26]: Removed

File: exasol_script_languages_container_tool/lib/api/upload.py (hunk 4 above)
Comment (tkilias) [reply exasol_script_languages_container_tool/lib/api/upload.py:64]: ok, then is fine

File: test/unit/cli/test_deploy.py (hunk 6 above)
Comment (tomuben) [reply test/unit/cli/test_deploy.py:45]: Done

File: test/unit/cli/test_deploy.py (hunk 7 above)
Comment (tomuben) [reply test/unit/cli/test_deploy.py:106]: Done

File: test/unit/test_click_api_consistency.py
Comment (tomuben) [reply test/unit/test_click_api_consistency.py:1]: Yes, I converted it to `pytest`.

**Review by tkilias — APPROVED** [review]

**Pull request merged**